Supporting multiple output formats

ferdicar opened the pull request:
The PlantUML class (and the plantuml server) supports various formats but it currently uses the `svg` format only.
With this change, the user can choose the output format using the `output_format` option in the configuration and selecting `svg` (default), `png` or `txt` (Ascii art)

Commit:
Supporting multiple output formats

The PlantUML class (and the plantuml server) supports various formats
but it currently uses the `svg` format only.
With this change, the user can choose the output format using the
`output_format` option in the configuration and selecting `svg`
(default), `png` or `txt` (Ascii art)

MikhailKravets commented:
Hi @ferdicar! Thank you for the contribution. My initial focus on svg was intentional, as it allows for adding interactive features to diagrams (please see the example in the [showcase](https://mikhailkravets.github.io/mkdocs_puml/#showcase)). I'm concerned that supporting `png` or `ascii` might compromise these interactive capabilities. 

Additionally, I’m not sure if a `png` can be seamlessly placed within a `div` without using an `img` tag, which may introduce further complexities. 

I appreciate your understanding and am happy to discuss this further if you have ideas for addressing these concerns!

ferdicar commented:
> Hi @ferdicar! Thank you for the contribution. My initial focus on svg was intentional, as it allows for adding interactive features to diagrams (please see the example in the [showcase](https://mikhailkravets.github.io/mkdocs_puml/#showcase)). I'm concerned that supporting `png` or `ascii` might compromise these interactive capabilities.
> 
> Additionally, I’m not sure if a `png` can be seamlessly placed within a `div` without using an `img` tag, which may introduce further complexities.
> 
> I appreciate your understanding and am happy to discuss this further if you have ideas for addressing these concerns!

Yes, you are perfectly right. As I saw that the PlantUML class accepted the format I assumed that it could support them. Definitely, png and ascii should be handled differently. I can propose a change to support them if you think it makes sense. In the meanwhile, I'd convert this PR in a draft.

MikhailKravets commented:
This functionality could potentially be integrated into the PlantUML class for standalone use. However, it appears that the library is not commonly used in a standalone context. I'll close this PR for now until further requests of this feature. Thanks again 🔥
